src/Features/Core/Portable/ChangeSignature/SignatureChange.cs

```diff
@@ -55,5 +58,66 @@ public SignatureChange(ParameterConfiguration originalConfiguration, ParameterCo
 
         internal SignatureChange WithoutAddedParameters()
             => new SignatureChange(OriginalConfiguration, UpdatedConfiguration.WithoutAddedParameters());
+
+        internal void LogTelemetry()
+        {
+            var originalListOfParameters = OriginalConfiguration.ToListOfParameters();
+            var updatedListOfParameters = UpdatedConfiguration.ToListOfParameters();
+
+            ChangeSignatureLogger.LogTransformationInformation(
+                numOriginalParameters: originalListOfParameters.Length,
+                numParametersAdded: updatedListOfParameters.Count(p => p is AddedParameter),
+                numParametersRemoved: originalListOfParameters.Count(p => !updatedListOfParameters.Contains(p)),
+                anyParametersReordered: AnyParametersReordered(originalListOfParameters, updatedListOfParameters));
+
+            foreach (var addedParameter in updatedListOfParameters.OfType<AddedParameter>())
+            {
+                if (addedParameter.IsRequired)
+                {
+                    ChangeSignatureLogger.LogAddedParameterRequired();
+                }
+
+                if (addedParameter.TypeBinds)
+                {
+                    ChangeSignatureLogger.LogAddedParameterTypeBinds();
+                }
+
+                if (addedParameter.IsCallsiteTodo)
+                {
+                    ChangeSignatureLogger.LogAddedParameter_ValueTODO();
+                }
+                else if (addedParameter.IsCallsiteOmitted)
+                {
+                    ChangeSignatureLogger.LogAddedParameter_ValueOmitted();
+                }
+                else
+                {
+                    if (addedParameter.UseNamedArguments)
```

Comment:
nit: flatten the nested if-else

Reply:
I like it this way (in the else) because if I pull it out to top-level it implies you can have UseNamedArguments set along with IsCallsite[Todo|Omitted], which it can't be. Or did you mean something else?